phpBB/download/file.php

```diff
@@ -11,310 +11,45 @@
-	$phpbb_class_loader_ext->register();
-
-	// Set up container
-	$phpbb_container_builder = new \phpbb\di\container_builder($phpbb_root_path, $phpEx);
-	$phpbb_container = $phpbb_container_builder->with_config($phpbb_config_php_file)->get_container();
-
-	$phpbb_class_loader->set_cache($phpbb_container->get('cache.driver'));
-	$phpbb_class_loader_ext->set_cache($phpbb_container->get('cache.driver'));
-
-	// set up caching
-	/* @var $cache \phpbb\cache\service */
-	$cache = $phpbb_container->get('cache');
-
-	/* @var $phpbb_dispatcher \phpbb\event\dispatcher */
-	$phpbb_dispatcher = $phpbb_container->get('dispatcher');
-
-	/* @var $request \phpbb\request\request_interface */
-	$request	= $phpbb_container->get('request');
-
-	/* @var $db \phpbb\db\driver\driver_interface */
-	$db			= $phpbb_container->get('dbal.conn');
-
-	/* @var $phpbb_log \phpbb\log\log_interface */
-	$phpbb_log	= $phpbb_container->get('log');
-
-	unset($dbpasswd);
-
-	/* @var $config \phpbb\config\config */
-	$config = $phpbb_container->get('config');
-
-	// load extensions
-	/* @var $phpbb_extension_manager \phpbb\extension\manager */
-	$phpbb_extension_manager = $phpbb_container->get('ext.manager');
-
-	// worst-case default
-	$browser = strtolower($request->header('User-Agent', 'msie 6.0'));
-
-	/* @var $phpbb_avatar_manager \phpbb\avatar\manager */
-	$phpbb_avatar_manager = $phpbb_container->get('avatar.manager');
-
-	if (@is_file($phpbb_root_path . $config['exts_composer_vendor_dir'] . '/autoload.php'))
-	{
-		require_once($phpbb_root_path . $config['exts_composer_vendor_dir'] . '/autoload.php');
-	}
-
-	$filename = $request->variable('avatar', '');
-	$avatar_group = false;
-	$exit = false;
-
-	if (isset($filename[0]) && $filename[0] === 'g')
-	{
-		$avatar_group = true;
-		$filename = substr($filename, 1);
-	}
-
-	// '==' is not a bug - . as the first char is as bad as no dot at all
-	if (strpos($filename, '.') == false)
-	{
-		send_status_line(403, 'Forbidden');
-		$exit = true;
-	}
-
-	if (!$exit)
-	{
-		$ext		= substr(strrchr($filename, '.'), 1);
-		$stamp		= (int) substr(stristr($filename, '_'), 1);
-		$filename	= (int) $filename;
-		$exit = set_modified_headers($stamp, $browser);
-	}
-	if (!$exit && !in_array($ext, array('png', 'gif', 'jpg', 'jpeg')))
-	{
-		// no way such an avatar could exist. They are not following the rules, stop the show.
-		send_status_line(403, 'Forbidden');
-		$exit = true;
-	}
-
+	$response = new RedirectResponse(
+		$controller_helper->route('phpbb_storage_avatar', array(
+			'file'	=> $request->variable('avatar', ''),
+		), false),
+		301
+	);
+	$response->send();
 
-	if (!$exit)
-	{
-		if (!$filename)
-		{
-			// no way such an avatar could exist. They are not following the rules, stop the show.
-			send_status_line(403, 'Forbidden');
-		}
-		else
-		{
-			send_avatar_to_browser(($avatar_group ? 'g' : '') . $filename . '.' . $ext, $browser);
-		}
-	}
-	file_gc();
+	exit;
```

Comment:
Isn't there something else we have to do before exiting a script?

Reply:
Not in this file, i left this from the original file.php